pkg/monitor/agent/agent.go

```diff
@@ -373,6 +384,9 @@ func (a *Agent) processPerfRecord(scopedLog *logrus.Entry, record perf.Record) {
 		})
 
 	} else {
+		if a.limiter != nil {
```

Comment:
What's the reason you chose to limit only the sample events? I'd expect an uptick in the number of lost samples, and maybe hence also in the number of calls to `notifyPerfEventsLostLocked` and `sendToListenersLocked`, which also contribute to CPU usage - did you observe this in your testing?

Reply:
Relatedly, this limiter does not limit agent events (c.f. `SendEvent`), is that by choice/what's the reasoning?
Granted, those are likely at a much smaller rate, but this should be documented at least.

Reply:
> What's the reason you chose to limit only the sample events? I'd expect an uptick in the number of lost samples, and maybe hence also in the number of calls to `notifyPerfEventsLostLocked` and `sendToListenersLocked`, which also contribute to CPU usage - did you observe this in your testing?

Yes. Indeed that was the exact behavior.

As I understand this the perfEvents are the ones directly related to workloads behavior and network traffic, hence they are expected to grow in an unwanted scenarios (misbehaving workloads, DoS attacks). That's why I originally decided to throttle only these events.

Limiting lostEvents will actually remove the visibility of processed and lost events and they are much less cpu time consuming than perfEvents. So generally I'm not really into this change. And for sure this should utilize another instance of rateLimiter then and be configurable with its own limit. In this case also an additional prometheus counter will be needed to have visibility into lostEvents throttled by rateLimiter.

The agentEvents from my understanding are not directly related to traffic but rather to changes in a cluster. In my test cases it was never a problem, but maybe we can introduce a similar solution for these. I would use another instance of the rate limiter for agent events though.

Reply:
I suggest changing the option's name if you don't handle agent event cases to clarify that the rate limit only applies to perf events.

Also, similar to what @bimmlerd suggested, we could apply this rate limit inside the BPF data path. So that events won't be ingested from starting point. As you said, filtering events in the lower place is better. Could you let me know if you have already considered that?